src/main/kotlin/assertk/assertions/inputstream.kt

```diff
@@ -0,0 +1,118 @@
+package assertk.assertions
+
+import assertk.Assert
+import assertk.assertions.support.expected
+import java.io.InputStream
+
+/** Asserts that the actual stream has the same content as the other stream.
+ *
+ * @param other whichs content is compared to the actual one
+ */
+fun Assert<InputStream>.hasSameContentAs(other: InputStream) {
+
+    var size = 0L
+
+    val actualBuffer = ByteArray(BUFFER_SIZE)
+    val otherBuffer = ByteArray(BUFFER_SIZE)
+
+    while (true) {
+        val actualRead = fillBuffer(actual, actualBuffer)
+        val otherRead = fillBuffer(other, otherBuffer)
+
+        if (actualRead == otherRead) {
+
+            if(actualRead==0) {
+                // the streams have the same size and are finished and have the same content
+                return
+            }
+            else {
+                val pos = compare(actualRead, actualBuffer, otherBuffer)
+                if(pos==null) {
+                    // the current buffers are equals and we can try the next block
+                    size += actualRead
+
+                } else {
+                    // the first difference is in position pos
+
+                    val actualSize = size + actualRead + consume(actual)
+                    val otherSize = size + otherRead + consume(other)
+
+                    expected("to have the same content, but actual stream differs at pos $size. Actual stream: value=0x${actualBuffer[pos].toHexString()}, size=$actualSize. Other stream: value=0x${otherBuffer[pos].toHexString()}, size=$otherSize")
+                }
+            }
+
+        } else {
+
+            if(actualRead==0) {
+                val actualSize = actualRead + size
+                val otherSize = consume(other) + otherRead + size
+                expected("to have the same size, but actual stream size (${actualSize}) differs from other stream size ($otherSize)")
+            }
+            else if (otherRead==0) {
+                val actualSize = consume(actual) + actualRead + size
+                val otherSize = otherRead + size
+                expected("to have the same size, but actual stream size ($actualSize) differs from other stream size (${otherSize})")
+            }
+            else {
+                val pos = compare(Math.min(actualRead, otherRead), actualBuffer, otherBuffer)
+                val actualSize = consume(actual) + actualRead + size
+                val otherSize = consume(other) + otherRead + size
+                if(pos==null) {
+                    expected("to have the same size, but actual size (${actualSize}) differs from other size (${otherSize})")
+                }
+                else {
+                    expected("to have the same content, but actual stream differs at pos $size. Actual stream: value=0x${actualBuffer[pos].toHexString()}, size=$actualSize. Other stream: value=0x${otherBuffer[pos].toHexString()}, size=$otherSize")
+                }
+            }
+        }
+    }
```

Comment:
Should you close the streams?

Reply:
Good question.

Usually I like that the code who creates the stream also closes is.
As they are not opened in the assertion method but in the test code, it should be closed there.
On the other hand it would lead to easier test code, if the test code does not need to close it.
 
What do you think?

Reply:
Hm good point. I totally agree with that sentiment. However, here the assertion always consumes all of both streams so there really isn't anything to do afterwords except close them. I also worry that this is something people will forget to do and not notice it's an issue until they have a lot of tests.

```kotlin
actualStream.use {
  expectedStream.use {
    assert(actualStream).hasSameContentAs(expectedStream)
  }
}
```
seems a bit verbose. Should definitely document that these assertions do close the streams.